packages/jaeger-ui/src/model/link-patterns.tsx

```diff
@@ -110,6 +110,27 @@ export function getParameterInAncestor(name: string, span: Span) {
     }
     currentSpan = getParent(currentSpan);
   }
+
+  return undefined;
+}
+
+const getValidTraceKeys = memoize(10)((trace: Trace) => {
+  const validKeys = (Object.keys(trace) as (keyof Trace)[]).filter(
+    key => typeof trace[key] === 'string' || typeof trace[key] === 'number'
+  );
+  return validKeys;
+});
+
+export function getParameterInTrace(name: string, trace: Trace | undefined) {
```

Comment:
it looks like the other similar functions return `{key, value}`, not just one value. Also, you later check `if(val)`, but it's possible that the value is 0 or false. Returning a dictionary vs. undefined would fix that as well.